### ISSUE TITLE
Styleint Treats Multiple Components As Same Component

### ISSUE THREAD

**Issue by Undistraction** [styled-components/stylelint-processor-styled-components#215]
The following two components defined in the same file results in an error:

**Components**
```javascript
const Layout = styled(HSpreadLayout)`
  > *:first-child {
    padding: 1.25rem;
  }
`

const NavList = styled(HList)`
  > * + * {
    border-left: 1px solid #bbb;
  }
`
```

**Error**
```
Expected selector ".selector6 > * + *" to come before selector ".selector5 > *:first-child"   no-descending-specificity
```

**Expected**
Each component should be evaluated separately.

- Node 8.11.3
- "stylelint": "^9.5.0",
- "stylelint-config-recommended": "^2.1.0",
- "stylelint-config-styled-components": "^0.1.1",
- "stylelint-processor-styled-components": "^1.3.2"

**Comment by Undistraction** [comment]
@chinesedfan That seems like a pretty big issue to me because it forces you to define your components in an order just to satisfy the processor. It could also potentially lead to cases that cannot be satisfied by changing component order.
